### PR TITLE
change cdk code to reflect final prod architecture, add prod github actions

### PR DESCRIPTION
**Pull request by JackVCurtis**

### CODE REVIEW

File: .github/workflows/deploy_app_prod.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        os: [ubuntu-latest]
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Use Node ${{ matrix.node }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Install deps
+        run: npm ci --legacy-peer-deps
```
Comment (mikestone14) [inline .github/workflows/deploy_app_prod.yaml]: Do we still need the legacy peer deps option?

File: .github/workflows/deploy_app_staging.yaml [hunk 2]
```diff
@@ -25,6 +25,7 @@ jobs:
         run: npm run test
   build-and-deploy:
     name: Build and Deploy Docker Image
+    needs: test
```
Comment (mikestone14) [inline .github/workflows/deploy_app_staging.yaml:28]: What does this option do?

**Review by mikestone14 — APPROVED** [review]
I’m not too familiar with this area of the code but overall looks good, just a couple small things.

File: .github/workflows/deploy_app_prod.yaml (hunk 1 above)
Comment (JackVCurtis) [reply .github/workflows/deploy_app_prod.yaml]: Good eye, I believe we don't

File: .github/workflows/deploy_app_staging.yaml (hunk 2 above)
Comment (JackVCurtis) [reply .github/workflows/deploy_app_staging.yaml:28]: Waits for tests to pass before starting the deploy. Previously these were running in parallel.

**Pull request merged**